pithos/pithos.py

```diff
@@ -1037,71 +919,49 @@ def on_response(dialog, response):
         dialog.connect('response', on_response)
         dialog.show()
 
-    def query_position(self):
-      pos_stat = self.player.query(self._query_position)
-      if pos_stat:
-        _, position = self._query_position.parse_position()
-        return position
-
-    def query_duration(self):
-      dur_stat = self.player.query(self._query_duration)
-      if dur_stat:
-        _, duration = self._query_duration.parse_duration()
-        return duration
-
-    def query_buffer(self):
-        buffer_stat = self.player.query(self._query_buffer)
-        if buffer_stat:
-            return self._query_buffer.parse_buffering_percent()[0]
-        else:
-            return True
-
-    def on_gst_stream_start(self, bus, message):
+    def on_player_new_stream_duration(self, player, duration):
         # Edge case. We might get this singal while we're reconnecting to Pandora.
         # If so self.current_song will be None.
         if self.current_song is None:
             return
         # Fallback to using song.trackLength which is in seconds and converted to nanoseconds
-        self.current_song.duration = self.query_duration() or self.current_song.trackLength * Gst.SECOND
+        self.current_song.duration = duration or self.current_song.trackLength * Gst.SECOND
```

Comment:
__[flake8]__
* [F405] 'Gst' may be undefined, or defined from star imports: .pandora, .pandora.data

<!-- Comment by [SideCI](https://sideci.com) -->